File: d01/d01.hs
module D01 where
import Data.List.Split

d01 :: Int -> IO ()
d01 part = do
    input <- readFile "1"
    print $ getAnswer (map (read::String->Int) (splitOn "\n" input)) part

part1 = 1::Int
part2 = 3::Int

getAnswer :: [Int] -> Int -> Int
getAnswer xs part = length . filter (\x -> fst x < snd x) . zip xs $ drop part xs

File: JS/d01/d01.js
import reader from '../readFile.js'

//d01
console.log((reader.getLines("d01").filter((a,b,c)=>a>+c[b-1]).length))

//d01-2
console.log((reader.getLines("d01").filter((a,b,c)=>a>+c[b-3]).length))